lua/lvim/config/_deprecated.lua

```diff
@@ -53,6 +53,22 @@ function M.handle()
   lvim.lsp.popup_border = {}
   setmetatable(lvim.lsp.popup_border, mt)
 
+  ---@deprecated
+  lvim.lsp.float = {}
+  setmetatable(lvim.lsp.float, {
+    __newindex = function(_, k, _)
+      deprecate("lvim.lsp.float." .. k, "Use `lvim.lsp.handlers` instead.")
+    end,
+  })
+
+  ---@deprecated
+  lvim.lsp.diagnostics = {}
+  setmetatable(lvim.lsp.diagnostics, {
+    __newindex = function(_, k, _)
+      deprecate("lvim.lsp.diagnostics." .. k, "Use `vim.diagnostic.config()` instead.")
```

Comment:
I think we should still maintain backwards compatibility here, but add the warning regardless.

https://github.com/LunarVim/LunarVim/blob/9ecad22dbb6ad4bbe1c02bf654ca92be59a56808/lua/lvim/lsp/handlers.lua#L6-L14